095-support-for-custom-authentication-type-in-kafka-clients.md

```diff
@@ -0,0 +1,42 @@
+# Support for Custom Authentication Type in Kafka Clients
+## Current situation
+Strimzi currently supports several standard authentication mechanisms for Kafka clients, including:
+- TLS-based authentication
+- SASL mechanisms such as `scram-sha-256`, `scram-sha-512`, `plain`, and `oauth`
+These mechanisms cover a wide range of use cases, but there is no built-in capability for users who need highly customized authentication logic.
+## Motivation
+Organizations may have unique security requirements that cannot be addressed by the existing authentication mechanisms. These requirements may involve integrating with proprietary security systems, leveraging custom token-based schemes, or handling complex credential management.
+By providing support for custom authentication types, Strimzi will enable users to supply their own authentication logic while maintaining compatibility with Kafka security standards.
+### Benefits:
+- Greater flexibility for organizations with custom security requirements.
+- Reduced need for workarounds when integrating with non-standard authentication systems.
+- Better alignment with enterprise security architectures.
+## Proposal
+### Introduction
+The proposal introduces a new Kafka client authentication type called `custom`. This type allows users to specify a fully qualified class name of a custom `AuthenticateCallbackHandler` implementation to handle their specific authentication logic.
+### Configuration
+The configuration for the custom authentication would look like this:
+```yaml
+authentication:
+  type: custom
+  customCallbackClass: com.example.kafka.auth.MyCustomAuthHandler
```

Comment:
other than just configuring, you should explain that a user has to bake an image by adding the jar including the handler class, right?